mmdet/models/task_modules/coders/bucketing_bbox_coder.py

```diff
@@ -85,11 +86,13 @@ def decode(self, bboxes, pred_bboxes, max_shape=None):
         cls_preds, offset_preds = pred_bboxes
         assert cls_preds.size(0) == bboxes.size(0) and offset_preds.size(
             0) == bboxes.size(0)
-        decoded_bboxes = bucket2bbox(bboxes, cls_preds, offset_preds,
-                                     self.num_buckets, self.scale_factor,
-                                     max_shape, self.clip_border)
-
-        return decoded_bboxes
+        bboxes, loc_confidence = bucket2bbox(bboxes, cls_preds, offset_preds,
+                                             self.num_buckets,
+                                             self.scale_factor, max_shape,
+                                             self.clip_border)
+        if self.with_boxlist:
+            bboxes = HorizontalBoxes(bboxes, clone=False)
```

Comment:
this is a hard code, can we select which boxlist we can use here?

Reply:
The box coder in mmdetection can only produce horizontal boxes. Because box coder can be replaced, we can create a new box code to process other types of boxes.